app/scripts/main.js

```diff
@@ -406,78 +403,60 @@ var iViz = (function(_, $, cbio, QueryByGeneUtil, QueryByGeneTextArea) {
       var def = new $.Deferred();
       var self = this;
       var data = self.getGroupNdx(groupId);
-
-      $.when(window.iviz.datamanager.getStudyToSampleToPatientdMap()).then(function(_studyToSampleToPatientMap) {
-        $.when(window.iviz.datamanager.getCaseLists()).then(function(_caseLists) {
-          var _sequencedCaseUIDs = [];
-          var _allCaseUIDs = [];
-          var _sequencedCaseUIdsMap = {};
+      
+      $.when(window.iviz.datamanager.getCnaFractionData(),
+        window.iviz.datamanager.getMutationCount())
+        .then(function(_cnaFractionData, _mutationCountData) {
+          var _hasCNAFractionData = _.keys(_cnaFractionData).length > 0;
+          var _hasMutationCountData = _.keys(_mutationCountData).length > 0;
           
-          $.each(_caseLists, function(studyId, caseList) {
-            if (caseList.sequencedSampleIds.length > 0) {
-              $.each(caseList.sequencedSampleIds, function(index, sampleId) {
-                _sequencedCaseUIDs.push(_studyToSampleToPatientMap[studyId].sample_to_uid[sampleId]);
-              });
-            }
-            if (caseList.allSampleIds.length > 0) {
-              $.each(caseList.allSampleIds, function(index, sampleId) {
-                _allCaseUIDs.push(_studyToSampleToPatientMap[studyId].sample_to_uid[sampleId]);
-              });
-            }
-          });
-          _sequencedCaseUIDs = _sequencedCaseUIDs.length > 0 ? _sequencedCaseUIDs : _allCaseUIDs;
-          _.each(_sequencedCaseUIDs, function(_sampleUid) {
-            _sequencedCaseUIdsMap[_sampleUid] = _sampleUid;
-          });
-
-          $.when(window.iviz.datamanager.getCnaFractionData(),
-            window.iviz.datamanager.getMutationCount())
-            .then(function(_cnaFractionData, _mutationCountData) {
-              var _hasCNAFractionData = _.keys(_cnaFractionData).length > 0;
-              var _hasMutationCountData = _.keys(_mutationCountData).length > 0;
-
-              if ((!_hasCNAFractionData || !_hasMutationCountData) && firstScatterShow) {
-                charts['MUT_CNT_VS_CNA'].show = false;
-                firstScatterShow = false;
-                _self.$dispatch('remove-chart', 'MUT_CNT_VS_CNA', groupId);//rearrange layout
-              }
+          // firstScatterShow is a global variable. If web page is just initialized, firstScatterShow is true and 
+          // it turns to false when cna.json or mutation.json returns null. When people want to add the empty 
+          // scatter chart, firstScatterChart(false) ensures the "remove empty chart" module won't be executed.
+          if ((!_hasCNAFractionData || !_hasMutationCountData) && firstScatterShow) { // remove empty scatter chart when web page just loaded
```

Comment:
Sorry, I didn't explain clearly. I don't want the dispatch event to be sent when people just add a chart. It should be only triggered when the page is loading. Otherwise, it will always remove the chart instead of generating an empty scatterplot.